zerver/actions/streams.py

```diff
@@ -1560,18 +1560,22 @@ def send_change_stream_message_retention_days_notification(
     with override_language(stream.realm.default_language):
         if old_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
             old_retention_period = _("Forever")
-            new_retention_period = f"{new_value} days"
-            summary_line = f"Messages in this stream will now be automatically deleted {new_value} days after they are sent."
+            new_retention_period = _("{retention_period} days").format(retention_period=new_value)
+            summary_line = _(
+                "Messages in this channel will now be automatically deleted {retention_period} days after they are sent."
+            ).format(retention_period=new_value)
         elif new_value == Stream.MESSAGE_RETENTION_SPECIAL_VALUES_MAP["unlimited"]:
-            old_retention_period = f"{old_value} days"
+            old_retention_period = _("{retention_period} days").format(retention_period=old_value)
             new_retention_period = _("Forever")
-            summary_line = _("Messages in this stream will now be retained forever.")
+            summary_line = _("Messages in this channel will now be retained forever.")
         else:
-            old_retention_period = f"{old_value} days"
-            new_retention_period = f"{new_value} days"
-            summary_line = f"Messages in this stream will now be automatically deleted {new_value} days after they are sent."
+            old_retention_period = _("{retention_period} days").format(retention_period=old_value)
+            new_retention_period = _("{retention_period} days").format(retention_period=new_value)
```

Comment:
Made these consistent so that there should only be one new string for translators. This way "day" will be translated to the user's language.

Reply:
That works for me. Updated for `number_of_days`.